src/services/ResourceService.js

```diff
@@ -243,6 +243,13 @@ async function init (currentUser, challengeId, resource, isCreated) {
   const handle = resource.memberHandle
   const memberId = await helper.getMemberIdByHandle(resource.memberHandle)
 
+  // check if the resource is reviewer role and has already made a submission in the challenge
+  const submissionsRes = await helper.getRequest(`${config.SUBMISSIONS_API_URL}`, { challengeId: challengeId, perPage: 100 })
```

Comment:
why are you fetching all submissions and then checking for the member id instead of simply passing the `memberId` property to the query and only receive submission for the particular member?

Reply:
Oh, let me double check the submission API, didn't know we have such API